user-service/src/services/scheduler/scheduler.service.ts

```diff
@@ -0,0 +1,68 @@
+import {IUserService} from "../user/user.service.interface";
+import {IRateService} from "../rate/rate.service.interface";
+import cron from 'node-cron'
+import { v4 as uuid } from 'uuid';
+import {serviceErrorHandler} from "../../error/handler/error.handler";
+import {IEmailDetails, IRateEmailMessageConfig} from "./scheduler.service.interface";
+import {IEvent, IQueueService} from "../queue/queue.service.interface";
+import userService from "../user/user.service";
+import rateService from "../rate/rate.service";
+import {rateMessageConfig} from "../../config/scheduler.config";
+import rabbitService from "../queue/rabbit.service";
+import {IEventService} from "../event/event.service.interface";
+import eventService from "../event/event.service";
+
+export class SchedulerService{
+
+    constructor(private userService: IUserService,
+                private rateService: IRateService,
+                private rateEmailMessageConfig: IRateEmailMessageConfig,
+                private queueService: IQueueService,
+                private eventService: IEventService) {}
+
+    public start() {
+        cron.schedule('0 12 * * *', () => {
+            this.sendEmails().catch((err) => {
+                    serviceErrorHandler(err)
+                }
+            );
+        });
+    }
+
+    private async sendEmails(): Promise<void> {
+        let emails: string[] = [];
+        let rate: number = 0;
+
+        try {
+            rate = await this.rateService.getExchangeRate();
+            emails = await this.userService.getAllUsersEmails();
+        } catch (err) {
```

Comment:
maybe there should be return statement here because the code of sendEmails method will continue execution after catch block and if rate service returned an error then emails will be sent with rate equals to 0 (your default value). Also you can wrap entire sendEmails method code with try catch then you will avoid such error.